src/community/community.dtos.ts

```diff
@@ -0,0 +1,252 @@
+import {
+    IsOptional,
+    IsArray,
+    IsString,
+    MaxLength,
+    IsNumberString,
+    IsMongoId,
+    IsNotEmpty,
+    ValidateNested, IsIn, IsEthereumAddress, ArrayNotEmpty
+} from 'class-validator';
+import { ApiProperty, OmitType, PartialType, IntersectionType } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import { ObjectId } from 'mongodb';
+import constants from '../common/constants';
+
+
+export abstract class CommunityAdministratorBase {
+    @IsNotEmpty()
+    @IsIn(Array.from(constants.PROVIDERS.keys()))
+    @ApiProperty({
+        description: 'Provider id',
+        required: true,
+    })
+        provider_id: string;
+}
+
+export class CommunityAdministratorDiscordDto extends CommunityAdministratorBase {
+
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord user IDs of the administrator',
+        required: false,
+        isArray: true,
+    })
+        user_allowlist: string[];
+    
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord role IDs of the administrator',
+        required: false,
+        isArray: true,
+    })
+        role_allowlist: string[];
+}
+
+export class CommunityAdministratorEthereumDto extends CommunityAdministratorBase {
+
+    @IsNotEmpty()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsEthereumAddress({ each: true })
+    @ApiProperty({
+        description: 'Ethereum addesses of the administrator',
+        required: true,
+        isArray: true,
+    })
+        address_allowlist: string[];
+}
+
+export abstract class CommunityClientConfigBase extends CommunityAdministratorBase {
+}
+
+export class CommunityClientConfigDiscordDto extends CommunityAdministratorBase {
+
+    @IsNotEmpty()
+    @IsNumberString()
+    @ApiProperty({
+        description: 'Discord server This config applies to',
+        required: true,
+    })
+        guild_id: string;
+
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord channel IDs of channels Governator is allowed to post in',
+        required: false,
+        isArray: true,
+    })
+        channel_allowlist: string[];
```

Comment:
If we have an entry in both the allowlist and denylist, which outcome is resolved?

Reply:
That part is not implemented yet but I think the denylist should probanly have priority. But now that you mention it, let me try if I can prevent duplicate entries during validation